### ISSUE TITLE
Bug: consistent-this: [‘error’, ‘app’] does not work

### ISSUE THREAD

**Issue by akobrina** [eslint/eslint#15701]
### Environment

Node version: v14.16.0
npm version: 7.8.0
Local ESLint version: v8.10.0
Global ESLint version: v8.10.0
Operating System: Mac OS Big Sur v. 11.5.1

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
.eslintrc.js
'use strict';

module.exports = {
  env: {
    browser: true,
    es6: true
  },
  plugins: [
    'eslint-plugin-html',
  ],
  extends: ['eslint:recommended'],
  rules: {
    'strict': ['error', 'global'],
    'no-redeclare': ['error', { builtinGlobals: true }],
    'dot-notation': 'error',
    'no-confusing-arrow': ['error'],
    'camelcase': ['error'],
    'quotes': ['error', 'single', { avoidEscape: true, allowTemplateLiterals: true }],
    'quote-props': ['error', 'consistent-as-needed'],
    'no-invalid-this': 'error',
    'consistent-this': ['error', 'app'],
    'eqeqeq': ['error', 'always'],
    'no-multi-assign': 'error',
    'no-array-constructor': 'error',
    'no-new-object': 'error',
    'no-new-wrappers': 'error',
    'yoda': 'error',
    'no-unneeded-ternary': 'error',
    'no-nested-ternary': 'error'
  },
};
```
</details>

```js
function Foo() {
    var app = this;
    app.a = 0;
}
```
When I use app alias, ESlint still flags it and says: Unexpected alias 'app' for 'this'.eslint[consistent-this]
I also tried to update the rule and use 'me' or 'self' instead of 'app' and got the same result. The only way that I found ESlint does not flag it is if I use 'that' in the rule: 'consistent-this': ['error', 'that'].

### What did you expect to happen?

I expected ESLint to not flag the use of 'app' alias when capturing the current execution context.

### What actually happened?

ESLint highlights 'app = this' with error and shows this message on hover: Unexpected alias 'app' for 'this'.eslint[consistent-this]

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @akobrina, thanks for the issue!

I can't reproduce this, here's online demo with your example: [demo link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGNvbnNpc3RlbnQtdGhpczogWydlcnJvcicsICdhcHAnXSAqL1xuXG5mdW5jdGlvbiBGb28oKSB7XG4gICAgdmFyIGFwcCA9IHRoaXM7XG4gICAgYXBwLmEgPSAwO1xufVxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

> ESLint highlights 'app = this' with error and shows this message on hover: Unexpected alias 'app' for 'this'.eslint[consistent-this]

Can you run ESLint from the command line and check if the same error will be reported? Did you try restarting your IDE after changing the `.eslintrc.js` config file?

**Comment by akobrina** [comment]
Hello @mdjermanovic, thank you for following up. I figured out what the problem was. While I had the 'consistent-this': ['error', 'app'], rule set in my .eslintrc.js file, what I was not aware of is that ESlint will actually pick up stuff from the comments with the JS files and evaluate those comments as linting rules. So, in my test file where I was doing some dummy code, I added explanatory comments that looked like this:

```
/*eslint consistent-this: ["error", "that" ]*/
//There are many commonly used aliases for 'this' such as that, self or me. This rule enforces that 'that' is the reserved alias for 'this'
function Foo() {
    var app = this;
    app.a = 0;
}
```
So, the comment above my test JS block was overwriting the eslint rule from the eslintrc.js file. I discovered it completely by accident. Maybe as an improvement, you could make something like this easier to troubleshoot by adding some kind of message that lets users know from where the ESlint rule originated? Maybe if a user is in --debug mode or make it more visible on the ESlint website that ESlint will evaluate comments in JS files?
